enumerables.rb

```diff
@@ -0,0 +1,135 @@
+# rubocop: disable Metrics/ModuleLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i <= size - 1
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each do |item|
+      selected.push(item) if yield(item)
+    end
+    selected
+  end
+
+  # rubocop: disable Style/CaseEquality, Style/IfInsideElse
+  def my_all?(var = nil)
+    result = true
+    my_each do |item|
+      if block_given?
+        result = false unless yield(item)
+      elsif var.nil?
+        result = false unless item
+      else
+        result = false unless var === item
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |item|
+      if block_given?
+        result = true if yield(item)
+      elsif var.nil?
+        result = true if item
+      else
+        result = true if var === item
+      end
+    end
+    result
+  end
+
+  def my_none?(var = nil)
+    result = true
+    my_each do |item|
+      if block_given?
+        result = false if yield(item)
+      elsif var.nil?
+        result = false if item
+      else
+        result = false if var === item
+      end
+    end
+    result
+  end
+
+  # rubocop: enable Style/CaseEquality, Style/IfInsideElse
+  def my_count(var = nil)
```

Comment:
enumerable #my_count if a range is given returns the number of items in self.
![Screen Shot 2020-06-27 at 3 04 14 PM](https://user-images.githubusercontent.com/52430951/85931115-7c13f580-b887-11ea-9fa0-91702c96cf89.png)